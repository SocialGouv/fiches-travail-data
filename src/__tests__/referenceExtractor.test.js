const fs = require("fs");
const path = require("path");
const { classifyTokens, extractReferences } = require("../referenceExtractor");
const { resolveReferences } = require("../referenceResolver");

const annotatedTokens = fs
  .readFileSync(path.join(__dirname, "referenceExtractor.test.txt"))
  .toString()
  .split("\n");

const tokens = [];
const labels = [];

annotatedTokens.map((line) => {
  const [t, l] = line.split("\t");
  tokens.push(t);
  labels.push(l);
});

const testCases = [
  {
    input:
      "les modalités fixées par les articles L. 2313‑8 et R. 2313-3 à R. 2313-6 du code du travail ainsi que le L. 1251-18",
  },
  {
    input: "L. 1251-23xx du code du travail",
  },
  {
    input:
      "l’allocation de remplacement pour maternité ou paternité, prévues aux articles L. 613-19 à L.613-19-2 et L. 722-8 à 25 du code de la sécurité sociale, aux articles L. 732-10 à L. 732-12-1 du code rural et à l’article 17 de la loi n° 97-1051 du 18 novembre 1997 d’orientation sur la pêche maritime et les cultures marines",
  },
  {
    input: `Article D212 du code penal et article R413`,
  },
  { input: `Article D212` },
  { input: `Article D-212` },
  { input: `Article D.212` },
  { input: `Article D212-3` },
  { input: `Article D-212-4` },
  { input: `Article X*212-4` },
  { input: `Article D. 212-4` },
  { input: `Article D.  212-4` },
  { input: `Article D.212-5` },
  { input: `Article D.212-5-6` },
  { input: `Article D.212-5-6-7` },
  { input: `Article XD212` },
];

test("should extract article tokens in examples", () => {
  expect(
    testCases.map(({ input }) => extractReferences(input))
  ).toMatchSnapshot();
});

test("should success with actual real life set", () => {
  const predictions = classifyTokens(tokens);
  // keep this to easily check prediction errors :
  /*
  const report = tokens.map((t, i) => {
    const label = labels[i];
    const prediction = predictions[i];
    const ok = prediction == label;
    return [
      `${ok ? "\x1b[0m" : "\x1b[31m"}`,
      `${t}\t${label}\t${prediction}\n`
    ];
  });
  console.log(...report.flat());
  */
  expect(predictions).toEqual(labels);
});

it("should find with code for actual real life set", () =>
  expect(extractReferences(tokens.join(" "))).toMatchSnapshot());

it("should resolve example codes", () => {
  const refs0 = extractReferences(testCases[0].input);
  expect(resolveReferences(refs0)).toMatchSnapshot();

  const refs1 = extractReferences(testCases[1].input);
  expect(resolveReferences(refs1)).toMatchSnapshot();
});

const rangeCases = [
  "L. 1251-21 à L. 1251-23xx du code du travail",
  "L. 1233‑34 à L. 1233-35-1 du code du travail",
  "L. 2312-72 à 2312-77 du code du travail",
  "L. 2312-72 à 2312-77 du code de l'éducation",
  "D. 5132-9 à D. 5132-10-4 du code du travail",
  "D. 5132-9 à D. 5132-10-4",
  "L. 2315-38 à 40 du code du travail",
  "L. 351-1 à L. 351-5 du code de la sécurité sociale",
];

it("should resolve ranges", () => {
  const refs = rangeCases.map((c) => {
    const extractedRefs = extractReferences(c);
    const resolvedRefs = resolveReferences(extractedRefs);
    return resolvedRefs;
  });

  expect(refs).toMatchSnapshot();
});
