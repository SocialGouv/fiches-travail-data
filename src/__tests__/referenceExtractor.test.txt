Le	O
nombre	O
et	O
le	O
périmètre	O
des	O
établissements	O
distincts	O
sont	O
déterminés	O
selon	O
les	O
modalités	O
fixées	O
par	O
les	O
articles	O
L.	B-ART
2313-8	B-ART
et	O
R.	B-ART
2313-3	B-ART
à	B-ART
R.	B-ART
2313-6	B-ART
du	O
code	O
du	O
travail	O
.	O
’	O
action	O
,	O
dans	O
les	O
conditions	O
fixées	O
par	O
l	O
’	O
article	O
R.	B-ART
2242-7	B-ART
du	O
code	O
du	O
travail.Dans	O
la	O
limite	O
me	O
La	O
protection	O
contre	O
les	O
discriminations	O
»	O
.Une	O
pénalité	O
financière	O
,	O
à	O
défaut	O
d	O
’	O
accord	O
ou	O
de	O
plan	O
d	O
’	O
actionDans	O
des	O
conditions	O
fixées	O
,	O
pour	O
l	O
’	O
essentiel	O
,	O
par	O
l	O
’	O
article	O
L.	B-ART
2242-8	B-ART
du	O
code	O
du	O
travail	O
et	O
par	O
les	O
articles	O
R.	B-ART
2242-2	B-ART
à	B-ART
R.	B-ART
2242-11	B-ART
,	O
à	O
défaut	O
d	O
’	O
accord	O
,	O
de	O
son	O
plan	O
d	O
’	O
action	O
relatif	O
à	O
l	O
’	O
égalité	O
professionnelle	O
entre	O
les	O
femmes	O
et	O
les	O
hommes	O
est	O
prévue	O
par	O
les	O
articles	O
L.	B-ART
2242-9	B-ART
et	O
R.	B-ART
2242-9	B-ART
à	B-ART
R.	B-ART
2242-11	B-ART
du	O
code	O
du	O
travail	O
.	O
cté	O
au	O
fonds	O
de	O
solidarité	O
vieillesse	O
mentionné	O
à	O
l	O
’	O
article	O
L.	B-ART
135-1	B-ART
du	O
code	O
de	O
la	O
sécurité	O
sociale.Cette	O
pénalité	O
pourra	O
également	O
être	O
appliquée	O
en	O
l	O
’	O
absence	O
de	O
publication	O
,	O
par	O
l	O
’	O
entreprise	O
,	O
de	O
l	O
’	O
«	O
Index	O
de	O
l	O
’	O
égalité	O
Femmes-Hommes	O
»	O
(	O
voir	O
ci-dessus	O
)	O
ou	O
aux	O
entreprises	O
qui	O
n	O
’	O
auront	O
pas	O
pris	O
les	O
mesures	O
correctives	O
en	O
cas	O
d	O
’	O
Index	O
inférieur	O
à	O
75	O
points	O
.	O
L	O
’	O
«	O
Index	O
de	O
l	O
’	O
égalité	O
Femmes-Hommes	O
»	O
est	O
établi	O
à	O
partir	O
des	O
indicateurs	O
figurant	O
aux	O
articles	O
D.	B-ART
1142-2	B-ART
(	O
entreprises	O
de	O
plus	O
de	O
250	O
salariés	O
)	O
et	O
D.	B-ART
1142-2-1	B-ART
du	O
code	O
du	O
travail	O
(	O
-	O
les	O
établissements	O
mentionnés	O
aux	O
articles	O
L.	B-ART
4521-1	B-ART
et	O
suivants	O
du	O
Code	O
du	O
travail	O
(	O
établissements	O
comprenant	O
au	O
moins	O
une	O
installation	O
nucléaire	O
de	O
base	O
,	O
classés	O
Seveso	O
ou	O
certains	O
gisements	O
miniers	O
)	O
.	O
ateurs	O
figurant	O
aux	O
articles	O
D.	B-ART
1142-2	B-ART
(	O
entreprises	O
de	O
plus	O
de	O
250	O
salariés	O
)	O
et	O
D.	B-ART
1142-2-1	B-ART
du	O
code	O
du	O
travail	O
(	O
entreprises	O
de	O
50	O
à	O
250	O
salariés	O
)	O
.	O
Il	O
est	O
publié	O
annuellement	O
,	O
au	O
plus	O
tard	O
le	O
1er	O
mars	O
de	O
l	O
’	O
année	O
en	O
cours	O
,	O
au	O
titre	O
de	O
l	O
’	O
année	O
précédente	O
,	O
sur	O
le	O
site	O
internet	O
de	O
l	O
’	O
entreprise	O
lorsqu	O
’	O
il	O
en	O
existe	O
un	O
.	O
À	O
défaut	O
,	O
il	O
est	O
porté	O
à	O
la	O
connaissance	O
des	O
salariés	O
par	O
tout	O
moye	O
Crash	O
:	O
es	O
employeurs	O
d	O
’	O
au	O
moins	O
11	O
salariés	O
(	O
seuil	O
applicable	O
à	O
compter	O
du	O
1er	O
janvier	O
2020	O
,	O
précédemment	O
fixé	O
à	O
10	O
salariés	O
)	O
doivent	O
effectuer	O
cette	O
transmission	O
à	O
Pôle	O
emploi	O
par	O
voie	O
électronique	O
(	O
sauf	O
impossibilité	O
pour	O
une	O
cause	O
qui	O
leur	O
est	O
étrangère	O
)	O
,	O
selon	O
les	O
modalités	O
fixées	O
par	O
l	O
’	O
arrêté	O
du	O
14	O
juin	O
2011	O
.	O
Pour	O
la	O
mise	O
en	O
œuvre	O
de	O
cette	O
obligation	O
de	O
transmission	O
par	O
voie	O
électronique	O
(	O
art	O
.	O
R.	B-ART
1234-9	B-ART
du	O
code	O
du	O
travail	O
)	O
,	O
l	O
’	O
effectif	O
salarié	O
et	O
les	O
règles	O
de	O
franchissement	O
de	O
seuils	O
d	O
’	O
effectif	O
sont	O
déterminés	O
selon	O
les	O
modalités	O
prévues	O
aux	O
articles	O
L.	B-ART
130-1	B-ART
et	O
R.	B-ART
130-1	B-ART
du	O
code	O
de	O
la	O
sécurité	O
sociale	O
,	O
sous	O
réserv	O
La	O
réponse	O
intervient	O
dans	O
les	O
30	O
jours	O
qui	O
suivent	O
sa	O
demande	O
et	O
contient	O
les	O
informations	O
mentionnées	O
à	O
l	O
’	O
article	O
D.	B-ART
8223-2	B-ART
du	O
Code	O
du	O
travail	O
.	O
Sur	O
ce	O
point	O
,	O
on	O
peut	O
consulter	O
les	O
informations	O
diffusées	O
sur	O
le	O
site	O
de	O
l	O
’	O
Urssaf	O
Une	O
copie	O
des	O
déclarations	O
de	O
détachement	O
mentionnées	O
aux	O
articles	O
R.	B-ART
1263-3	B-ART
,	O
R.	B-ART
1263-4	B-ART
et	O
R.	B-ART
1263-6	B-ART
du	O
code	O
du	O
travail	O
est	O
annexée	O
au	O
registre	O
unique	O
du	O
personnel	O
et	O
rendue	O
accessible	O
aux	O
délégués	O
du	O
personnel	O
(	O
ou	O
aux	O
membres	O
de	O
la	O
délégation	O
du	O
personnel	O
du	O
comité	O
social	O
et	O
économique	O
lorsque	O
cette	O
instance	O
de	O
représentation	O
du	O
personnel	O
aura	O
été	O
mise	O
en	O
place	O
dans	O
l	O
’	O
entreprise	O
)	O
et	O
aux	O
fonctionnaires	O
et	O
agents	O
chargés	O
de	O
veiller	O
à	O
l	O
’	O
application	O
du	O
code	O
du	O
tra	O
l	O
’	O
allocation	O
de	O
remplacement	O
pour	O
maternité	O
ou	O
paternité	O
,	O
prévues	O
aux	O
articles	O
L.	B-ART
613-19	B-ART
à	B-ART
L.	B-ART
613-19-2	B-ART
et	O
L.	B-ART
722-8	B-ART
à	B-ART
L.	B-ART
722-8-3	B-ART
du	O
code	O
de	O
la	O
sécurité	O
sociale	O
,	O
aux	O
articles	O
L.	B-ART
732-10	B-ART
à	B-ART
L.	B-ART
732-12-1	B-ART
du	O
code	O
rural	O
et	O
à	O
l	O
’	O
article	O
17	O
de	O
la	O
loi	O
n°	O
97-1051	O
du	O
18	O
novembre	O
1997	O
d	O
’	O
orientation	O
sur	O
la	O
pêche	O
maritime	O
et	O
les	O
cultures	O
marines	O
lui-même	O
pris	O
sur	O
la	O
base	O
de	O
l	O
’	O
article	O
L.	B-ART
4412-2	B-ART
,	O
issu	O
de	O
la	O
loi	O
n°	O
2016-1088	O
du	O
8	O
août	O
2016	O
6	O
domaines	O
d	O
’	O
activité	O
défini	O
par	O
le	O
décret	O
du	O
9	O
mai	O
2017	O
(	O
art	O
.	O
R.	B-ART
4412-97/II	B-ART
)	O
,	O
la	O
Direction	O
générale	O
du	O
travail	O
(	O
DGT	O
)	O
a	O
choisi	O
de	O
s	O
’	O
appuyer	O
sur	O
un	O
travail	O
de	O
normalisation	O
.	O
En	O
l	O
’	O
absence	O
de	O
tout	O
accord	O
(	O
décision	O
unilatérale	O
de	O
l	O
’	O
employeur	O
)	O
,	O
ces	O
modalités	O
sont	O
définies	O
par	O
le	O
règlement	O
intérieur	O
du	O
comité	O
,	O
prévu	O
par	O
l	O
’	O
article	O
L.	B-ART
2315-24	B-ART
du	O
code	O
du	O
travail.Dans	O
tous	O
les	O
cas	O
,	O
les	O
dispositions	O
d	O
’	O
ordre	O
public	O
relatives	O
à	O
la	O
composition	O
,	O
aux	O
missions	O
et	O
au	O
fonctionnement	O
de	O
la	O
commission	O
,	O
prévues	O
aux	O
articles	O
L.	B-ART
2315-38	B-ART
à	B-ART
40	B-ART
,	O
devront	O
être	O
respectées	O
.	O
``	O
,	O
Les	O
effectifs	O
s	O
’	O
apprécient	O
dans	O
le	O
cadre	O
de	O
l	O
’	O
entreprise	O
ou	O
dans	O
le	O
cadre	O
de	O
chaque	O
établissement	O
distinct	O
,	O
conformément	O
aux	O
dispositions	O
prévues	O
par	O
le	O
Code	O
du	O
travail	O
et	O
Article	O
L1111-2	B-ART
.	O
La	O
délégation	O
du	O
personnel	O
comporte	O
un	O
nombre	O
égal	O
de	O
titulaires	O
et	O
de	O
suppléants	O
.	O
satisfaire	O
aux	O
conditions	O
de	O
majorité	O
prévues	O
aux	O
articles	O
L.	B-ART
2314-6	B-ART
du	O
code	O
du	O
travail	O
(	O
double	O
condition	O
de	O
majorité	O
,	O
voir	O
précisions	O
ci-dessus	O
)	O
.	O
Pour	O
sa	O
part	O
,	O
l	O
’	O
accord	O
d	O
’	O
entreprise	O
(	O
ou	O
de	O
groupe	O
)	O
autorisant	O
le	O
recours	O
au	O
vote	O
électronique	O
(	O
à	O
défaut	O
d	O
’	O
accord	O
,	O
l	O
’	O
employeur	O
peut	O
décider	O
de	O
ce	O
recours	O
)	O
est	O
soumis	O
aux	O
seules	O
conditions	O
de	O
validité	O
prévues	O
par	O
le	O
code	O
du	O
code	O
du	O
travail	O
dans	O
le	O
cadre	O
du	O
droit	O
commun	O
de	O
la	O
négociation	O
collective.Conception	O
et	O
mise	O
en	O
place	O
du	O
dispositifLa	O
conception	O
et	O
la	O
mise	O
en	O
place	O
du	O
système	O
de	O
vote	O
électronique	O
peuvent	O
être	O
confiées	O
à	O
un	O
prestataire	O
choisi	O
par	O
l	O
’	O
employeur	O
sur	O
la	O
base	O
d	O
’	O
un	O
cahier	O
des	O
charges	O
respectant	O
les	O
dispositions	O
des	O
articles	O
R.	B-ART
2314-6	B-ART
et	O
suivants	O
du	O
code	O
du	O
travail	O
.	O
nomique	O
et	O
sociale	O
ainsi	O
que	O
les	O
modalités	O
de	O
son	O
fonctionnement	O
,	O
notamment	O
les	O
droits	O
d	O
’	O
accès	O
,	O
sont	O
ceux	O
mentionnés	O
à	O
l	O
’	O
article	O
L	B-ART
2312-36	B-ART
du	O
Code	O
du	O
travail	O
.	O
Ces	O
éléments	O
sont	O
précisés	O
:	O
par	O
les	O
articles	O
R.	B-ART
2312-8	B-ART
et	O
R.	B-ART
2312-10	B-ART
à	B-ART
R.	B-ART
2312-14	B-ART
pour	O
les	O
entreprises	O
de	O
moins	O
de	O
300	O
salariés	O
;	O
par	O
les	O
articles	O
R.	B-ART
2312-9	B-ART
et	O
R.	B-ART
2312-10	B-ART
à	B-ART
R.	B-ART
2312-14	B-ART
pour	O
les	O
entreprises	O
d	O
’	O
au	O
moins	O
300	O
salariés	O
.	O
Cette	O
base	O
de	O
données	O
,	O
régulièrement	O
mise	O
à	O
jour	O
,	O
est	O
accessible	O
en	O
permanence	O
aux	O
membres	O
de	O
la	O
délégation	O
du	O
personnel	O
du	O
CSE	O
ainsi	O