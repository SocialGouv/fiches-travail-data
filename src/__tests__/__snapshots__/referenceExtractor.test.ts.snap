// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should extract article tokens from "Article D.  212-4" 1`] = `
Array [
  Object {
    "code": undefined,
    "text": "D. 212-4",
  },
]
`;

exports[`should extract article tokens from "Article D. 212-4" 1`] = `
Array [
  Object {
    "code": undefined,
    "text": "D. 212-4",
  },
]
`;

exports[`should extract article tokens from "Article D.212" 1`] = `
Array [
  Object {
    "code": undefined,
    "text": "D.212",
  },
]
`;

exports[`should extract article tokens from "Article D.212-5" 1`] = `
Array [
  Object {
    "code": undefined,
    "text": "D.212-5",
  },
]
`;

exports[`should extract article tokens from "Article D.212-5-6" 1`] = `
Array [
  Object {
    "code": undefined,
    "text": "D.212-5-6",
  },
]
`;

exports[`should extract article tokens from "Article D.212-5-6-7" 1`] = `
Array [
  Object {
    "code": undefined,
    "text": "D.212-5-6-7",
  },
]
`;

exports[`should extract article tokens from "Article D-212" 1`] = `
Array [
  Object {
    "code": undefined,
    "text": "D-212",
  },
]
`;

exports[`should extract article tokens from "Article D-212-4" 1`] = `
Array [
  Object {
    "code": undefined,
    "text": "D-212-4",
  },
]
`;

exports[`should extract article tokens from "Article D212 du code penal et article R413" 1`] = `
Array [
  Object {
    "code": undefined,
    "text": "R413",
  },
]
`;

exports[`should extract article tokens from "Article D212" 1`] = `
Array [
  Object {
    "code": undefined,
    "text": "D212",
  },
]
`;

exports[`should extract article tokens from "Article D212-3" 1`] = `
Array [
  Object {
    "code": undefined,
    "text": "D212-3",
  },
]
`;

exports[`should extract article tokens from "Article X*212-4" 1`] = `Array []`;

exports[`should extract article tokens from "Article XD212" 1`] = `Array []`;

exports[`should extract article tokens from "L. 1251-23xx du code du travail" 1`] = `
Array [
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "L. 1251-23xx",
  },
]
`;

exports[`should extract article tokens from "les modalités fixées par les articles L. 2313‑8 et R. 2313-3 à R. 2313-6 du code du travail ainsi que le L. 1251-18" 1`] = `
Array [
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "L. 2313‑8",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "R. 2313-3 à R. 2313-6",
  },
  Object {
    "code": undefined,
    "text": "L. 1251-18",
  },
]
`;

exports[`should extract article tokens from "l’allocation de remplacement pour maternité ou paternité, prévues aux articles L. 613-19 à L.613-19-2 et L. 722-8 à 25 du code de la sécurité sociale, aux articles L. 732-10 à L. 732-12-1 du code rural et à l’article 17 de la loi n° 97-1051 du 18 novembre 1997 d’orientation sur la pêche maritime et les cultures marines" 1`] = `
Array [
  Object {
    "code": Object {
      "id": "LEGITEXT000006073189",
      "name": "code de la sécurité sociale",
    },
    "text": "L. 613-19 à L.613-19-2",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006073189",
      "name": "code de la sécurité sociale",
    },
    "text": "L. 722-8 à 25",
  },
]
`;

exports[`should find with code for actual real life set 1`] = `
Array [
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "L. 2313-8",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "R. 2313-3 à R. 2313-6",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "R. 2242-7",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "L. 2242-8",
  },
  Object {
    "code": undefined,
    "text": "R. 2242-2 à R. 2242-11",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "L. 2242-9",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "R. 2242-9 à R. 2242-11",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "D. 1142-2",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "D. 1142-2-1",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "L. 4521-1",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "D. 1142-2",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "D. 1142-2-1",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "R. 1234-9",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "D. 8223-2",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "R. 1263-3",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "R. 1263-4",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "R. 1263-6",
  },
  Object {
    "code": undefined,
    "text": "L. 4412-2",
  },
  Object {
    "code": undefined,
    "text": "R. 4412-97/II",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "L. 2315-24",
  },
  Object {
    "code": undefined,
    "text": "L. 2315-38 à 40",
  },
  Object {
    "code": undefined,
    "text": "L1111-2",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "L. 2314-6",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "R. 2314-6",
  },
  Object {
    "code": Object {
      "id": "LEGITEXT000006072050",
      "name": "code du travail",
    },
    "text": "L 2312-36",
  },
  Object {
    "code": undefined,
    "text": "R. 2312-8",
  },
  Object {
    "code": undefined,
    "text": "R. 2312-10 à R. 2312-14",
  },
  Object {
    "code": undefined,
    "text": "R. 2312-9",
  },
  Object {
    "code": undefined,
    "text": "R. 2312-10 à R. 2312-14",
  },
]
`;

exports[`should resolve example codes 1`] = `
Object {
  "LEGITEXT000006072050": Object {
    "articles": Array [
      Object {
        "fmt": "L2313-8",
        "id": "LEGIARTI000036262496",
        "text": "L. 2313‑8",
      },
      Object {
        "fmt": "R2313-3",
        "id": "LEGIARTI000039624940",
        "text": "R. 2313-3",
      },
      Object {
        "fmt": "R2313-4",
        "id": "LEGIARTI000036414723",
      },
      Object {
        "fmt": "R2313-5",
        "id": "LEGIARTI000039347715",
      },
      Object {
        "fmt": "R2313-6",
        "id": "LEGIARTI000039624937",
        "text": "R. 2313-6",
      },
      Object {
        "fmt": "L1251-18",
        "id": "LEGIARTI000006901269",
        "text": "L. 1251-18",
      },
    ],
    "name": "code du travail",
  },
}
`;

exports[`should resolve example codes 2`] = `
Object {
  "LEGITEXT000006072050": Object {
    "articles": Array [
      Object {
        "fmt": "L1251-23",
        "id": "LEGIARTI000006901276",
        "text": "L. 1251-23xx",
      },
    ],
    "name": "code du travail",
  },
}
`;

exports[`should resolve ranges 1`] = `
Array [
  Object {
    "LEGITEXT000006072050": Object {
      "articles": Array [
        Object {
          "fmt": "L1251-21",
          "id": "LEGIARTI000006901274",
          "text": "L. 1251-21",
        },
        Object {
          "fmt": "L1251-22",
          "id": "LEGIARTI000006901275",
        },
        Object {
          "fmt": "L1251-23",
          "id": "LEGIARTI000006901276",
        },
      ],
      "name": "code du travail",
    },
  },
  Object {
    "LEGITEXT000006072050": Object {
      "articles": Array [
        Object {
          "fmt": "L1233-34",
          "id": "LEGIARTI000036762068",
          "text": "L. 1233‑34",
        },
        Object {
          "fmt": "L1233-35",
          "id": "LEGIARTI000036261799",
        },
        Object {
          "fmt": "L1233-35-1",
          "id": "LEGIARTI000035643013",
          "text": "L. 1233-35-1",
        },
      ],
      "name": "code du travail",
    },
  },
  Object {
    "LEGITEXT000006072050": Object {
      "articles": Array [
        Object {
          "fmt": "L2312-72",
          "id": "LEGIARTI000035611295",
          "text": "L. 2312-72",
        },
        Object {
          "fmt": "L2312-73",
          "id": "LEGIARTI000035611297",
        },
        Object {
          "fmt": "L2312-74",
          "id": "LEGIARTI000035611299",
        },
        Object {
          "fmt": "L2312-75",
          "id": "LEGIARTI000035611301",
        },
        Object {
          "fmt": "L2312-76",
          "id": "LEGIARTI000035611303",
        },
        Object {
          "fmt": "L2312-77",
          "id": "LEGIARTI000035611305",
          "text": "2312-77",
        },
      ],
      "name": "code du travail",
    },
  },
  Object {},
  Object {
    "LEGITEXT000006072050": Object {
      "articles": Array [
        Object {
          "fmt": "D5132-10-1",
          "id": "LEGIARTI000029775717",
        },
        Object {
          "fmt": "D5132-10-2",
          "id": "LEGIARTI000029775713",
        },
        Object {
          "fmt": "D5132-10-3",
          "id": "LEGIARTI000029775708",
        },
        Object {
          "fmt": "D5132-10-4",
          "id": "LEGIARTI000029775703",
          "text": "D. 5132-10-4",
        },
      ],
      "name": "code du travail",
    },
  },
  Object {
    "LEGITEXT000006072050": Object {
      "articles": Array [
        Object {
          "fmt": "D5132-10-1",
          "id": "LEGIARTI000029775717",
        },
        Object {
          "fmt": "D5132-10-2",
          "id": "LEGIARTI000029775713",
        },
        Object {
          "fmt": "D5132-10-3",
          "id": "LEGIARTI000029775708",
        },
        Object {
          "fmt": "D5132-10-4",
          "id": "LEGIARTI000029775703",
          "text": "D. 5132-10-4",
        },
      ],
      "name": "code du travail",
    },
  },
  Object {
    "LEGITEXT000006072050": Object {
      "articles": Array [
        Object {
          "fmt": "L2315-38",
          "id": "LEGIARTI000035626459",
          "text": "L. 2315-38",
        },
        Object {
          "fmt": "L2315-39",
          "id": "LEGIARTI000036262434",
        },
        Object {
          "fmt": "L2315-40",
          "id": "LEGIARTI000035626463",
          "text": "40",
        },
      ],
      "name": "code du travail",
    },
  },
  Object {
    "LEGITEXT000006073189": Object {
      "articles": Array [
        Object {
          "fmt": "L351-1",
          "id": "LEGIARTI000023035688",
          "text": "L. 351-1",
        },
        Object {
          "fmt": "L351-1-1",
          "id": "LEGIARTI000028498678",
        },
        Object {
          "fmt": "L351-1-2",
          "id": "LEGIARTI000025014540",
        },
        Object {
          "fmt": "L351-1-3",
          "id": "LEGIARTI000028498841",
        },
        Object {
          "fmt": "L351-1-4",
          "id": "LEGIARTI000035640851",
        },
        Object {
          "fmt": "L351-2",
          "id": "LEGIARTI000031686263",
        },
        Object {
          "fmt": "L351-3",
          "id": "LEGIARTI000028498769",
        },
        Object {
          "fmt": "L351-3-1",
          "id": "LEGIARTI000031667793",
        },
        Object {
          "fmt": "L351-4",
          "id": "LEGIARTI000037063369",
        },
        Object {
          "fmt": "L351-4-1",
          "id": "LEGIARTI000017845585",
        },
        Object {
          "fmt": "L351-4-2",
          "id": "LEGIARTI000028497884",
        },
        Object {
          "fmt": "L351-5",
          "id": "LEGIARTI000021537902",
          "text": "L. 351-5",
        },
      ],
      "name": "code de la sécurité sociale",
    },
  },
]
`;
